clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -71,6 +71,8 @@
   private final Map<Class<? extends EhcacheEntityResponse>, List<ResponseListener<? extends EhcacheEntityResponse>>> responseListeners =
     new ConcurrentHashMap<>();
   private final List<DisconnectionListener> disconnectionListeners = new CopyOnWriteArrayList<>();
+  private volatile boolean initCompleted = false;
+  private final Object initLock = new Object();
```

Comment:
Can't we use the existing lock or any other private object in this class? And ok, I accept the solution with a notify :-)

Reply:
There is a lock object..but its purpose was different..did not want to mix concerns that might cause problems later..

Reply:
Ok. I agree me way is more hacky